### PR TITLE
BAU - fix to prevent modified container being added to the end of the…

### PR DESCRIPTION
**Pull request by SteveSugden**
… container list

### CODE REVIEW

File: app/controllers/declaration/SealController.scala [hunk 1]
```diff
@@ -159,7 +159,10 @@ class SealController @Inject()(
       val updatedContainers =
         model.containerData
           .map(_.containers)
-          .fold(Seq(container))(_.filterNot(_.id == container.id) :+ container)
+          .fold(Seq(container))(_.map {
+            case c if (c.id == container.id) => container
+            case x                           => x
```
[the remark this answers is not in the document]
Comment (SteveSugden) [reply app/controllers/declaration/SealController.scala]: Good point.  And I've removed the unnecessary () before Ed gets on my case again!

File: app/controllers/declaration/SealController.scala [hunk 2]
```diff
@@ -153,13 +153,16 @@ class SealController @Inject()(
...
 
-  private def updateCache(container: Container)(implicit req: JourneyRequest[AnyContent]) =
+  private def updateCache(updatedContainer: Container)(implicit req: JourneyRequest[AnyContent]) =
     updateExportsDeclarationSyncDirect(model => {
 
       val updatedContainers =
         model.containerData
           .map(_.containers)
-          .fold(Seq(container))(_.filterNot(_.id == container.id) :+ container)
+          .fold(Seq(updatedContainer))(_.map {
```
Comment (bambuchaAdm) [inline app/controllers/declaration/SealController.scala]: Please move map logic for Containers object

Comment (SteveSugden) [reply app/controllers/declaration/SealController.scala]: Good idea.

Comment (SteveSugden) [reply app/controllers/declaration/SealController.scala]: Can you take a 2nd look @bambuchaAdm

**Pull request merged**